src/app/components/client/homepage.tsx

```diff
@@ -57,12 +108,21 @@ const Homepage = ({ data, batch }: { data: HomepageData; batch: Array<HomepageEn
         </section>
         <div className=''>
           <h2 className='text-xl md:text-4xl font-semibold pb-[60px]'>All Activity</h2>
-          <>
+          <div>
             {memoizedBatches?.map((entry, idx) => (
               <Post key={`${entry.id}_${idx}`} entry={entry} isActivePost={false} />
             ))}
-          </>
+          </div>
         </div>
+        <section>
+          <button
+            className='border-2 border-black p-6 py-2 flex items-center justify-center text-lg font-medium'
+            style={{ cursor: loading ? "not-allowed" : "pointer", pointerEvents: loading ? "none" : "visible" }}
```

Comment:
We do this with tailwind css. It'll reduce the extra style declaration here. You just need to add a ternary to the tailwind className